fvm/src/syscalls/actor.rs

```diff
@@ -10,32 +10,30 @@ pub fn resolve_address(
     context: Context<'_, impl Kernel>,
     addr_off: u32, // Address
     addr_len: u32,
-) -> Result<sys::out::actor::ResolveAddress> {
+) -> Result<u64> {
     let addr = context.memory.read_address(addr_off, addr_len)?;
-    let (resolved, value) = match context.kernel.resolve_address(&addr)? {
-        Some(id) => (0, id),
-        None => (-1, 0),
-    };
-    Ok(sys::out::actor::ResolveAddress { resolved, value })
+    let actor_id = context
+        .kernel
+        .resolve_address(&addr)?
+        .ok_or_else(|| syscall_error!(NotFound; "actor not found"))?;
```

Comment:
Returns an error on resolution failure, not a -1.

Reply:
I mean, just commenting that we're masking a potential NotFound from the kernel, but that it currently can't produce it. Definitely an uber-nit, not important.